src/latlon.py

```diff
@@ -12,14 +12,18 @@ def getlatlon(imgpath):
     im = rasterio.open(imgpath)
     crs = im.crs.__str__()  # Coordinate reference system code
     nrows, ncols = im.shape
-    rows, cols = np.meshgrid(np.arange(nrows), np.arange(ncols))
+    cols, rows = np.meshgrid(np.arange(ncols), np.arange(nrows))
     xs, ys = rasterio.transform.xy(im.transform, rows, cols)
+    xs = np.array(xs)
+    ys = np.array(ys)
+    
     # X and Y are the 1D index vectors
-    X = np.array(xs)[:, 0]
-    Y = np.array(ys)[0, :]
+    X = np.array(xs)[0, :]
+    Y = np.array(ys)[:, 0]
```

Comment:
Because of the earlier casting, those operations could be dropped here or the earlier ones removed.

Reply:
Done, this line was supposed to get changed due to the cast earlier.

Reply:
I'll add the test. Thanks for the data!